src/cpu/flexcpu/simple_dataflow_cpu.cc

```diff
@@ -597,7 +672,11 @@ void
 SimpleDataflowCPU::DataPort::recvReqRetry()
 {
     receivedPushback = false;
-    cpu->attemptAllMemReqs();
+
+    if (!cpu->attemptAllMemReqsEvent.scheduled()) {
+        cpu->schedule(&cpu->attemptAllMemReqsEvent,
+            cpu->clockedMemoryRequest ? cpu->clockEdge(Cycles(0)) : curTick());
+    }
```

Comment:
I would add comments to each of these points, mentioning the exact function that these events are going to call, like attemptAllMemReqs(), since that will make control flow easier to follow.